### DISCUSSION TITLE
Add Opensource / Selfhosted Provider

### DISCUSSION THREAD

**Discussion by mymanga** [vitodeploy/vito#332] [Ideas]
Would like to see support for Opensource / Selfhosted Providers like Cloudstack, Proxmox

Cloudstack API: https://cloudstack.apache.org/api/apidocs-4.19/
Proxmox API: https://pve.proxmox.com/wiki/Proxmox_VE_API

**Comment by hcaz** [comment]
This would be a great addition that sets apart from the hosted solutions. (+1 Proxmox for me)

**Comment by saeedvaziry** [comment]
I suggest developing a plugin for this

https://vitodeploy.com/docs/plugins
